.github/workflows/dxcint.yml

```diff
@@ -0,0 +1,69 @@
+name: dxCint tests
+on:
+  push:
+    branches:
+      - main
+      - develop
+  pull_request:
+    branches:
+      - main
+      - develop
+    types:
+      - opened
+      - synchronize
+      - ready_for_review
+      - labeled
+  workflow_dispatch:
+permissions:
+  contents: read
+  actions: write
+  packages: read
+jobs:
+  test:
+    name: dxCint Tests (Staging)
+    if: contains(github.event.pull_request.labels.*.name, 'dxcint') ||  github.event_name == 'workflow_dispatch' 
+    runs-on: ubuntu-20.04
+    env:
+      PROJECT: dxCompiler_playground
+      TEST_DIR: dxcint_testing
+    steps:
+      - name: Git checkout
+        uses: actions/checkout@v2
+      # performance optimizations https://abbbi.github.io/actions/
+      - uses: abbbi/github-actions-tune@v1
+      - name: Install java
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+      - name: Install dxpy and other dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y apt-transport-https wget git openssh-server tree python3 python3-pip python3-venv
+          cwltool=`jq -r '.execDepends.cwl[0].name' scripts/bundled_dependencies.json`
+          pip3 install --upgrade pip setuptools pyOpenSSL pytest
+          pip3 install termcolor dxpy==0.323.0 $cwltool
+          # install dxda
+          dxda_version=`jq -r '.dxda' scripts/bundled_dependencies.json`
+          sudo wget -O/usr/local/bin/dx-download-agent https://github.com/dnanexus/dxda/releases/download/${dxda_version}/dx-download-agent-linux
+          sudo chmod +x /usr/local/bin/dx-download-agent
```

Comment:
`dxcint` is a poetry package, where you have `poetry.lock` file with all dependency versions pinned - pretty handy and reproducible. Could you rewrite this bit with something like I did for [Scientific Product CI/CD](https://github.com/dnanexus/pseudopod/blob/ab0ba32a89216b18b02ebbcf3b4f41594ac52bd8/.github/workflows/sciprodbuild_unit_tests.yml#L44-L51)?

Also, not sure there's a need for `dx-download-agent`
The rest is perfect